_posts/2018-06-02-GraphQL-Stitching.markdown

```diff
@@ -0,0 +1,384 @@
+---
+layout: epic
+title: GraphQL Stitching 101
+date: 2018-11-21
+author: [orta]
+categories: [graphql, stitching, metaphysics]
+css: graphql-stitch
+comment_id: 450
+---
+
+Micro-Services make sense. You can scope a domain of your business to particular small unit of abstraction like an
+API. Doing so makes it easy to work in isolation, experiment with new ideas and evolve in many directions.
+
+We've been [carefully pushing][monolith] for years to move away from our single monolithic API, to a collection of
+smaller, more focused projects. Our [highlights docs][high] showcase this well. The movement to smaller composable
+services works great from an isolated platform/systems perspective but can be a bit tricky to handle with front-end
+clients. Until 2018, the way that we've addressed the growing complexity in our service later has been to migrate
+the complexity inside our main GraphQL API, [metaphysics][mp]. Metaphysics is our GraphQL API gateway that
+consolidates many API sources into a single service, then extending and interleaving and their data to make clients
+easier to write.
+
+However, as more services have been created, and grown - so has metaphysics. This creates a worrying trend, as the
+growth of code in metaphysics isn't quite linear.
+
+Our main line-of-thought on how to address this is via GraphQL schema stitching. We've been [running
+experiments][ex] in stitching for over a year, and have have been running with stitching enabled in production for
+a few months.
+
+<!-- more -->
+
+## What is Schema Stitching?
+
+The core idea behind schema stitching is that because GraphQL talks in type systems, you should be able to merge
+type systems from many GraphQL APIs into a single source of truth. Schema Stitching came out at the [end of
+2017][stitching_out] via the [`graphql-tools`][tools] and became very production-[ready in April
+2018][stitching_announcement].
+
+We started experimenting last year and would occasionally run it on staging to discover edge case issues. This
+meant the state of the project would ebb & flow between a blocked, or no-one having the bandwidth to work on it.
+This was fine, because our aim was [incremental evolutions over bold revolution][rev].
+
+Before we dive into implementation details, here's a quick glossary of terms before we start:
+
+- **GraphQL Schema** - a representation of your GraphQL's type system, containing all types and fields on them
+- **GraphQL Resolver** - every field accessed in a query has a corresponding resolver
```

Comment:
"GraphQL Resolver: every field accessed in a query has a corresponding resolver" -> 
"GraphQL Resolver: every field accessed in a query resolves a to a corresponding value" (or something) 

(Defining resolver but at the same time referencing resolver to support definition)

Reply:
good point